pkg/arvo/sys/vane/clay.hoon

```diff
@@ -2689,36 +2705,138 @@
+        work
+      ?>  ?=(%nako p.r.u.rut)
+      =/  nako  ;;(nako q.r.u.rut)
+      =/  missing  (missing-blobs nako)
+      =.  need.sat  `(list lobe)`(welp need.sat ~(tap in missing))
+      =.  nako.sat  (~(put to nako.sat) ~ nako)
+      work
+    ::
+    ++  missing-blobs
+      |=  =nako
+      ^-  (set lobe)
+      =/  yakis  ~(tap in lar.nako)
+      |-  ^-  (set lobe)
+      =*  yaki-loop  $
+      ?~  yakis
+        ~
+      =/  lobes=(list [=path =lobe])  ~(tap by q.i.yakis)
+      |-  ^-  (set lobe)
+      =*  blob-loop  $
+      ?~  lobes
+        yaki-loop(yakis t.yakis)
+      ?:  (~(has by lat.ran) lobe.i.lobes)
+        blob-loop(lobes t.lobes)
+      (~(put in blob-loop(lobes t.lobes)) lobe.i.lobes)
+    ::
+    ::  Receive backfill response
+    ::
+    ++  take-backfill
+      |=  =blob
+      ^+  ..abet
+      ?:  lost  ..abet
+      =?    need.sat
+          ?&  ?=(%delta -.blob)
+              !(~(has by lat.ran) q.q.blob)
+              !(~(has by have.sat) q.q.blob)
+          ==
+        [q.q.blob need.sat]
+      ::  We can't put a blob in lat.ran if its parent isn't already
+      ::  there.  Unions are in reverse order so we don't overwrite
+      ::  existing blobs.
+      ::
+      =.  ..abet
+        ?:  &(?=(%delta -.blob) !(~(has by lat.ran) q.q.blob))
+          ..abet(have.sat (~(uni by (malt [p.blob `^blob`blob] ~)) have.sat))
+        ..abet(lat.ran (~(uni by (malt [p.blob blob] ~)) lat.ran))
+      work(busy.sat |)
+    ::
+    ::  Fetch next blob
+    ::
+    ++  work
+      ^+  ..abet
+      ?:  busy.sat
+        ..abet
+      |-  ^+  ..abet
+      ?:  =(~ need.sat)
+        ::  NB: if you change to release nakos as we get enough blobs
+        ::  for them instead of all at the end, you *must* store the
+        ::  `lim` that should be applied after the nako is complete and
+        ::  not use the one in the rave, since that will apply to the
+        ::  end of subscription.
+        ::
+        =.  lat.ran  (~(uni by have.sat) lat.ran)
+        |-  ^+  ..abet
+        ?:  =(~ nako.sat)
+          ..abet
+        =^  next=(unit nako)  nako.sat  ~(get to nako.sat)
+        ?~  next
+          ..abet(done &)
+        =.  ..abet  (apply-foreign-update u.next)
+        =.  ..foreign-update  =<(?>(?=(^ ref) .) wake)
+        $
+      ?>  ?=(^ need.sat)
+      ::  This is what removes an item from `need`.  This happens every
+      ::  time we take a backfill response, but it could happen more than
+      ::  once if we somehow got this data in the meantime (maybe from
+      ::  another desk updating concurrently, or a previous update on this
+      ::  same desk).
+      ::
+      ?:  ?|  (~(has by lat.ran) i.need.sat)
+              (~(has by have.sat) i.need.sat)
+          ==
+        $(need.sat t.need.sat)
+      ::  Otherwise, fetch the next blob
+      ::
+      =/  =fill  [syd i.need.sat]
+      =/  =wire  /back-index/(scot %p her)/[syd]/(scot %ud inx)
```

Comment:
Yes each request is a new flow.  This is not new behavior.  The argument for putting them on separate flows is that some requests may be quite long while others are quick.  Consider if you request a large desk and, while it's downloading, you decide to request its `%cz` hash.  There's no reason those should interfere with each other.

Reply:
All else being equal, sure. But flows are permanent, and this seems like unbounded proliferation. Since it's not strictly necessary, I'd be inclined to limit them to `/our/her/desk`. It's not new behavior, but it's worth nailing down these norms at some point.